pkg/controller/scylladbmonitoring/sync_grafana.go

```diff
@@ -140,9 +140,24 @@ func makeGrafanaConfigs(sm *scyllav1alpha1.ScyllaDBMonitoring) (*corev1.ConfigMa
 }
 
 func makeGrafanaDashboards(sm *scyllav1alpha1.ScyllaDBMonitoring) (*corev1.ConfigMap, string, error) {
-	return grafanav1alpha1assets.GrafanaDashboardsConfigMapTemplate.RenderObject(map[string]any{
-		"scyllaDBMonitoringName": sm.Name,
-	})
+	if sm.Spec.Type == nil {
+		// This should have a default value, but it can be nil due to a version skew.
+		scyllaDBMonitoringTypeSAAS := scyllav1alpha1.ScyllaDBMonitoringTypeSAAS
+		sm.Spec.Type = &scyllaDBMonitoringTypeSAAS
```

Comment:
this will break the informers, you can't write to the shared cache. pls make a variable for the type to switch on

Reply:
Fixed. Btw should I add a test for `makeGrafanaDashboards`?

Reply:
I don't mind wither way but I'd not hold this over it

Reply:
Let's go ahead then